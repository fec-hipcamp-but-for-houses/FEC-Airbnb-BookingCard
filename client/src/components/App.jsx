import React from 'react';
import axios from 'axios';
import Booking from './Booking.jsx';
import Report from './Report.jsx';
import ReportModal from './ReportModal.jsx';
import Slider from './Slider.jsx';
import StarRating from './StarRating.jsx';

const body = {
  boxSizing: 'border-box',
  position: 'sticky',
  top: '20px',
  margin: '0 auto',
  zIndex: '3',
  fontFamily: 'Arial, Helvetica, sans-serif',
  color: '#484848',
};

const card = {
  width: '326px',
  paddingLeft: '24px',
  paddingRight: '24px',
  border: '1px solid #e4e4e4',
  backgroundColor: '#ffffff',
};

const contentContainer = {
  marginTop: '16px',
  marginBottom: '24px',
};

const priceAndRating = {
  borderBottom: '1px solid #e4e4e4',
  marginBottom: '16px',
};

const priceContainer = {
  
};

const reviewContainer = {
  marginBottom: '16px',
};

const baseRate = {
  fontSize: '22px',
  fontWeight: '800',
  lineHeight: '1.444444em',
};

const font12 = {
  fontSize: '12px',
};

const container = {
  marginLeft: '45px',
};

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = { listing: {}, hideSlide: true, hideReport: true };
  }

  componentDidMount() {
    this.getListing(192)
      .then(({ data }) => this.setState({ listing: data[0] }));

    window.addEventListener('scroll', this.handleScroll);
    document.addEventListener('mousedown', this.handleClick, false);
  }

  handleClick = (e) => {
    const modal = document.getElementById('modal');
    if (e.target === modal) this.closeReportModal();
  }

  onReportClick = () => { this.setState({ hideReport: false }); }

  closeReportModal = () => { this.setState({ hideReport: true }); }

  handleScroll = () => {
    const header = document.getElementsByClassName('fakeHeader')[0];
    const headerBottom = header.getBoundingClientRect().bottom;

    if (headerBottom < 0) {
      this.setState({ hideSlide: false });
    } else {
      this.setState({ hideSlide: true });
    }
  }

  getListing = listingId => axios.get('/listing', { params: { listingId } });

  render() {
    const { listing, hideSlide, hideReport } = this.state;

    return (
      <div style={body}>
        <div style={container}>
          <div style={card}>
            <div style={contentContainer}>
              <div style={priceAndRating}>
                <div style={priceContainer}>
                  <span style={baseRate}>{`$${listing.baseRate}`}</span>
                  <span style={font12}>per night</span>
                </div>
                <div style={reviewContainer}>
                  <StarRating stars={listing.averageRating} numberOfRatings={listing.numberOfRatings} />
                </div>
              </div>
              <Booking maxGuests={listing.maxGuests} maxInfants={listing.maxInfants} />
              <Slider hidden={hideSlide} />
            </div>
          </div>
          <Report onReportClick={this.onReportClick} />
          <div ref={(node) => { this.node = node; }}>
            {!hideReport && <ReportModal closeModal={this.closeReportModal} />}
          </div>
        </div>
      </div>
    );
  }
}

export default App;
