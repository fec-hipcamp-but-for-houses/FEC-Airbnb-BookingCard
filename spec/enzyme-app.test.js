import React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../client/src/components/App.jsx';

describe('App Component Test Suite', () => {
  it('Should render correctly with no props', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should hide Slider compoenent on render by default', () => {
    const wrapper = shallow(<App />);
    const hideSlideState = wrapper.state().hideSlide;
    expect(hideSlideState).toBe(true);
  });

  it('Should be able to open guest form with a click', () => {
    const wrapper = mount(<App />);
    wrapper
      .find('button#guestButton')
      .simulate('click');

    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('Should be able to render Report Modal with a click', () => {
    const wrapper = mount(<App />);
    wrapper
      .find('button#reportButton')
      .simulate('click');

    expect(wrapper.state('hideReport')).toEqual(false);
    wrapper.unmount();
  });
});
