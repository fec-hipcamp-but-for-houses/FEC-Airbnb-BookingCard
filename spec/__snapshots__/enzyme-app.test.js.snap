// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test Suite Should be able to open guest form with a click 1`] = `ReactWrapper {}`;

exports[`App Component Test Suite Should render correctly with no props 1`] = `ShallowWrapper {}`;
